scripts/interpreter.js

```diff
@@ -210,12 +217,20 @@
         intptr.startInterpreters = [];
         let interpreters = intptr.generateInterpreters(robots, codes, ball);
 
+        var step;
         // Set the starting positions for all robots
-        for (let interpreter of intptr.startInterpreters) {
+        for (let r = 0; r < robots.length; r++) {
             let i = 0;
-            while (!intptr.stopAll && i < 1000) {
-                interpreter.step();
-                i++;
+            if (!intptr.interpreters[r]) continue;
+            try {
+                while (!intptr.stopAll && i < 1000) {
+                    step = intptr.interpreters[r].step();
+                    i++;
+                }
+            } catch (e) {
+                if (step) {
```

Comment:
Not sure whats going on here, if there was a previous step, add error, but if there wasn't, dont do anything?

Reply:
I think it might not be necessary - but yea, it should only show the error if there was a previous step.